R/rdeephaven/R/client_wrapper.R

```diff
@@ -1,110 +1,249 @@
+      } else {
+        stop(paste0("'auth_type' must be 'anonymous', 'basic', or 'custom', but got ", auth_type, " instead."))
       }
-      if (table_object_class[[1]] == "tbl_df") {
-        return(TableHandle$new(private$tibble_to_dh_table(table_object)))
-      } else if (table_object_class[[1]] == "RecordBatchReader") {
-        return(TableHandle$new(private$rbr_to_dh_table(table_object)))
-      } else if ((length(table_object_class) == 4 &&
-        table_object_class[[1]] == "Table" &&
-        table_object_class[[3]] == "ArrowObject")) {
-        return(TableHandle$new(private$arrow_to_dh_table(table_object)))
+    }
+
+    # set session type if a valid session type is provided
+    if ((session_type == "python") || (session_type == "groovy")) {
+      options$set_session_type(session_type)
+    } else {
+      stop(paste0("'session_type' must be 'python' or 'groovy', but got ", session_type, " instead."))
+    }
+
+    # if tls is requested, set it and set the root_certs if provided
+    if (use_tls != FALSE) {
+      if (use_tls == TRUE) {
+        options$set_use_tls()
+        if (tls_root_certs != "") {
+          verify_string("tls_root_certs", tls_root_certs, TRUE)
+          options$set_tls_root_certs(tls_root_certs)
+        }
       } else {
-        stop(paste0("'table_object' must be either an R Data Frame, a dplyr Tibble, an Arrow Table, or an Arrow Record Batch Reader. Got an object of class ", table_object_class[[1]], " instead."))
+        stop(paste0("'use_tls' must be TRUE or FALSE, but got ", use_tls, " instead."))
       }
-    },
-
-    #' @description
-    #' Runs a script on the server. The script must be in the language that the server console was started with.
-    #' @param script Code to be executed on the server, passed as a string.
-    run_script = function(script) {
-      .verify_string("script", script)
-      private$internal_client$run_script(script)
     }
-  ),
-  private = list(
-    internal_client = NULL,
-    check_for_table = function(name) {
-      return(private$internal_client$check_for_table(name))
-    },
-    rbr_to_dh_table = function(rbr) {
-      ptr <- private$internal_client$new_arrow_array_stream_ptr()
-      rbr$export_to_c(ptr)
-      return(private$internal_client$new_table_from_arrow_array_stream_ptr(ptr))
-    },
-    arrow_to_dh_table = function(arrow_tbl) {
-      rbr <- as_record_batch_reader(arrow_tbl)
-      return(private$rbr_to_dh_table(rbr))
-    },
-    tibble_to_dh_table = function(tibbl) {
-      arrow_tbl <- arrow_table(tibbl)
-      return(private$arrow_to_dh_table(arrow_tbl))
-    },
-    df_to_dh_table = function(data_frame) {
-      arrow_tbl <- arrow_table(data_frame)
-      return(private$arrow_to_dh_table(arrow_tbl))
+
+    if (int_option != "") {
+      verify_string("int_option", int_option, TRUE)
+      new_int_option <- strsplit(int_option, ":", fixed = TRUE)
+      options$add_int_option(new_int_option[1], as.numeric(new_int_option[2]))
     }
```

Comment:
This looks wacky:
1) doing weird string handling stuff again
2) this is handling integer options as strings
3) this looks like multiple options can be provided, but this R API is not supporting it.  Why not a map containing the options?

Reply:
1. Following Python
2. Yes, weird
3. Good catch, I forgot about that